### ISSUE TITLE
Clicking on material of glTF export with no material assigned causes a crash

### ISSUE THREAD

**Issue by fazil47** [godotengine/godot#57955]
### Godot version

v4.0.alpha.custom_build [73f131208]

### System information

Windows 11, Vulkan Clustered, Nvidia 1650

### Issue description

Export a mesh instance with a mesh which hasn't been assigned a material as a glTF file. Godot crashes when you click on the material on the import dialog of the exported glTF. This doesn't happen in the alpha 2 build or in a Vulkan mobile project.

![image](https://user-images.githubusercontent.com/18116695/153539604-74fa8fbf-39e3-4631-871f-6c476256a773.png)

```
CrashHandlerException: Program crashed
Engine version: Godot Engine v4.0.alpha.custom_build (73f13120867fed5a2e71599e33f08222bc7a140e)
Dumping the backtrace. Please include this when reporting the bug on https://github.com/godotengine/godot/issues
[0] Object::get_class_name (C:\Users\fazil\repos\my_repos\godot\development\godot\core\object\object.h:725)
[1] Object::emit_signal (C:\Users\fazil\repos\my_repos\godot\development\godot\core\object\object.cpp:1048)
[2] Object::emit_signal (C:\Users\fazil\repos\my_repos\godot\development\godot\core\object\object.cpp:1160)
[3] TreeItem::set_collapsed (C:\Users\fazil\repos\my_repos\godot\development\godot\scene\gui\tree.cpp:540)
[4] TreeItem::uncollapse_tree (C:\Users\fazil\repos\my_repos\godot\development\godot\scene\gui\tree.cpp:551)
[5] SceneImportSettings::_select (C:\Users\fazil\repos\my_repos\godot\development\godot\editor\import\scene_import_settings.cpp:644)
[6] SceneImportSettings::_material_tree_selected (C:\Users\fazil\repos\my_repos\godot\development\godot\editor\import\scene_import_settings.cpp:697)
[7] call_with_variant_args_helper<SceneImportSettings> (C:\Users\fazil\repos\my_repos\godot\development\godot\core\variant\binder_common.h:237)
[8] call_with_variant_args<SceneImportSettings> (C:\Users\fazil\repos\my_repos\godot\development\godot\core\variant\binder_common.h:347)
[9] CallableCustomMethodPointer<SceneImportSettings>::call (C:\Users\fazil\repos\my_repos\godot\development\godot\core\object\callable_method_pointer.h:105)
[10] Callable::call (C:\Users\fazil\repos\my_repos\godot\development\godot\core\variant\callable.cpp:51)
[11] Object::emit_signal (C:\Users\fazil\repos\my_repos\godot\development\godot\core\object\object.cpp:1105)
[12] Object::emit_signal (C:\Users\fazil\repos\my_repos\godot\development\godot\core\object\object.cpp:1160)
[13] Tree::select_single_item (C:\Users\fazil\repos\my_repos\godot\development\godot\scene\gui\tree.cpp:2282)
[14] Tree::select_single_item (C:\Users\fazil\repos\my_repos\godot\development\godot\scene\gui\tree.cpp:2320)
[15] Tree::propagate_mouse_event (C:\Users\fazil\repos\my_repos\godot\development\godot\scene\gui\tree.cpp:2543)
[16] Tree::propagate_mouse_event (C:\Users\fazil\repos\my_repos\godot\development\godot\scene\gui\tree.cpp:2704)
[17] Tree::gui_input (C:\Users\fazil\repos\my_repos\godot\development\godot\scene\gui\tree.cpp:3389)
[18] Control::_call_gui_input (C:\Users\fazil\repos\my_repos\godot\development\godot\scene\gui\control.cpp:940)
[19] Viewport::_gui_call_input (C:\Users\fazil\repos\my_repos\godot\development\godot\scene\main\viewport.cpp:1293)
[20] Viewport::_gui_input_event (C:\Users\fazil\repos\my_repos\godot\development\godot\scene\main\viewport.cpp:1523)
[21] Viewport::push_input (C:\Users\fazil\repos\my_repos\godot\development\godot\scene\main\viewport.cpp:2690)
[22] Window::_window_input (C:\Users\fazil\repos\my_repos\godot\development\godot\scene\main\window.cpp:969)
[23] call_with_variant_args_helper<Window,Ref<InputEvent> const &,0> (C:\Users\fazil\repos\my_repos\godot\development\godot\core\variant\binder_common.h:232)
[24] call_with_variant_args<Window,Ref<InputEvent> const &> (C:\Users\fazil\repos\my_repos\godot\development\godot\core\variant\binder_common.h:347)
[25] CallableCustomMethodPointer<Window,Ref<InputEvent> const &>::call (C:\Users\fazil\repos\my_repos\godot\development\godot\core\object\callable_method_pointer.h:105)
[26] Callable::call (C:\Users\fazil\repos\my_repos\godot\development\godot\core\variant\callable.cpp:51)
[27] DisplayServerWindows::_dispatch_input_event (C:\Users\fazil\repos\my_repos\godot\development\godot\platform\windows\display_server_windows.cpp:1982)
[28] DisplayServerWindows::_dispatch_input_events (C:\Users\fazil\repos\my_repos\godot\development\godot\platform\windows\display_server_windows.cpp:1956)
[29] Input::_parse_input_event_impl (C:\Users\fazil\repos\my_repos\godot\development\godot\core\input\input.cpp:658)
[30] Input::flush_buffered_events (C:\Users\fazil\repos\my_repos\godot\development\godot\core\input\input.cpp:878)
[31] DisplayServerWindows::process_events (C:\Users\fazil\repos\my_repos\godot\development\godot\platform\windows\display_server_windows.cpp:1703)
[32] OS_Windows::run (C:\Users\fazil\repos\my_repos\godot\development\godot\platform\windows\os_windows.cpp:629)
[33] widechar_main (C:\Users\fazil\repos\my_repos\godot\development\godot\platform\windows\godot_windows.cpp:168)
[34] _main (C:\Users\fazil\repos\my_repos\godot\development\godot\platform\windows\godot_windows.cpp:190)
[35] main (C:\Users\fazil\repos\my_repos\godot\development\godot\platform\windows\godot_windows.cpp:204)
[36] WinMain (C:\Users\fazil\repos\my_repos\godot\development\godot\platform\windows\godot_windows.cpp:218)
[37] __scrt_common_main_seh (d:\a01\_work\6\s\src\vctools\crt\vcstartup\src\startup\exe_common.inl:288)
[38] BaseThreadInitThunk
-- END OF BACKTRACE --
```

### Steps to reproduce

- In a new scene add a `MeshInstance3D` node
- Assign it a mesh but not a material
- Export the scene as glTF
- Open the exported glTF in the import dialog
- Click on the material in the `Materials` tab

**Comment by akien-mga** [comment]
Fixed by #57969.

**Comment by fazil47** [comment]
#57969 doesn't fix this but #58189 will.
